### ISSUE TITLE
Reloading save transports all out-of-system cargo to player

### ISSUE THREAD

**Issue by Bladewood** [endless-sky/endless-sky#1624]
Flying a berserker and had four freighters in my fleet.
Freighters and (probably) my own berserker were carrying injured Syndicate miners to Sunracer.
Landed on Crossroads in Ruchbah while freighters were out of system.
Tried to depart.

> If you take off now you will fail a mission due to not having enough bunks available for 14 of the passengers. Are you sure you want to continue?
> `Cancel` `OK`

**Comment by tehhowch** [comment]
As mentioned on https://github.com/endless-sky/endless-sky/issues/1945#issuecomment-269917034, it seems like making a `uuid` class would be beneficial for resolving this and related issues.

(For those wondering, missions are currently referenced via a memory address that is only available when the game is loaded, and is not guaranteed to be consistent across save loads. In order to tell ships that they have mission passengers or mission cargo - which may not be uniquely named for a particular mission - the ships would need to associate a used cargo space slot / bunk with the owning mission. 
A UUID that is generated and assigned to a mission upon instantiation could replace this memory address, and would be persistent across save loads. The reason this and related issues are taking a while to fix is that there are a number of methods that rely on having access to the memory address, and not a string representation of that mission).

**Comment by downRightCoder256** [comment]
> You could also solve the issue by making mission descriptions in the save files "know" which star systems its cargo is in just like it already knows which star systems mission NPCs are in.

I'd say this would be better done by having the cargo be associated with specific ships, in a similar way outfits are.

**Comment by NomadicVolcano** [comment]
See bugfix #5070 which works like this:

> I'd say this would be better done by having the cargo be associated with specific ships, in a similar way outfits are.

**Comment by fuzzyTew** [comment]
Not sure why my search for 'saved game' didn't turn up this issue; I then opened #5461 instead.  Important information to add here is that the work was done to fix this issue in PR #5070 but that PR has since been closed by the submitter when they closed their github account.  Nomadic-Volcano left right as they were asked whether or not htey could make tests for it, and now the proposal in the PR is to merge a whole test framework in from a different fork.  How thorough tests are needed for work that attempts to improve behavior that is already pretty buggy?
